conda/core/initialize.py

```diff
@@ -1149,6 +1193,58 @@ def init_long_path(target_path):
                   'the fall 2016 "Anniversary update" or newer.')
             return Result.NO_CHANGE
 
+def _powershell_profile_content(conda_prefix):
+    if on_win:
+        conda_exe = join(conda_prefix, 'Scripts', 'conda.exe')
+    else:
+        conda_exe = join(conda_prefix, 'bin', 'conda')
+
+    conda_powershell_module = dals("""
+    #region conda initialize
+    # !! Contents within this block are managed by 'conda init' !!
+    (& {conda_exe} shell.powershell hook) | Out-String | Invoke-Expression
+    #endregion
+    """.format(conda_exe=conda_exe))
+
+    return conda_powershell_module
+
+def init_powershell_user(target_path, conda_prefix):
+    # target_path: $PROFILE
+    profile_path = target_path
+
+    # NB: the user may not have created a profile. We need to check
+    #     if the file exists first.
+    if os.path.exists(profile_path):
+        with open(profile_path) as fp:
+            profile_content = fp.read()
+    else:
+        profile_content = ""
+
+    profile_original_content = profile_content
+
+    # Find what content we need to add.
+    conda_initialize_content = _powershell_profile_content(conda_prefix)
+
+    # TODO: comment out old ipmos and Import-Modules.
+
+    if "#region conda initialize" not in profile_content:
+        profile_content += "\n{}\n".format(conda_initialize_content)
+
+    if profile_content != profile_original_content:
+        if context.verbosity:
+            print('\n')
+            print(target_path)
+            print(make_diff(profile_original_content, profile_content))
+        if not context.dry_run:
+            # Make the directory if needed.
+            if not os.path.exists(os.path.dirname(profile_path)):
+                os.makedirs(os.path.dirname(profile_path))
```

Comment:
For consistency with the rest of the code base, use

```python
            if not exists(dirname(profile_path)):
                mkdir_p(dirname(profile_path))
```

They're all already imported.

Reply:
Fixed.